### PR TITLE
[Completed][PyTorch] Add GAN

### PR DESCRIPTION
**Pull request by StevenJokes**
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by StevenJokes** [comment]
Please review: @AnirudhDagar 
Please help me `d2l build lib`

File: chapter_preface/index.md [hunk 1]
```diff
@@ -265,10 +265,9 @@ to be saved in the package, we will mark it with
 The `d2l` package is light-weight and only requires
 the following packages and modules as dependencies:
 
-```{.python .input  n=1}
+```{.python .input}
 #@tab all
-#@save
-import collections
+import collections  #@save
```
Comment (AnirudhDagar) [inline chapter_preface/index.md:270]: Don't change this.

File: chapter_preface/index.md [hunk 2]
```diff
@@ -336,16 +332,14 @@ to update your code and runtime environment.
 Here is how we import modules from TensorFlow.
 :end_tab:
 
-```{.python .input  n=1}
-#@save
-from mxnet import autograd, context, gluon, image, init, np, npx
+```{.python .input}
+from mxnet import autograd, context, gluon, image, init, np, npx  #@save
 from mxnet.gluon import nn, rnn
```
Comment (AnirudhDagar) [inline chapter_preface/index.md:337]: Don't change this.

File: chapter_preface/index.md [hunk 3]
```diff
@@ -336,16 +332,14 @@ to update your code and runtime environment.
...
 ```
 
-```{.python .input  n=1}
+```{.python .input}
 #@tab pytorch
-#@save
-import numpy as np
+import numpy as np  #@save
```
Comment (AnirudhDagar) [inline chapter_preface/index.md:342]: Don't change this.

File: chapter_preface/index.md [hunk 4]
```diff
@@ -354,10 +348,9 @@ from torch.utils import data
 from torchvision import transforms
 ```
 
-```{.python .input  n=1}
+```{.python .input}
 #@tab tensorflow
-#@save
-import numpy as np
+import numpy as np  #@save
```
Comment (AnirudhDagar) [inline chapter_preface/index.md:353]: Don't change this.

**Comment by StevenJokes** [comment]
OK? now

**Comment by mli** [comment]
Job d2l-en/PR-1358/5 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1358/

**Comment by mli** [comment]
Job d2l-en/PR-1358/7 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1358/

**Comment by StevenJokes** [comment]
Please review: @AnirudhDagar

**Comment by StevenJokes** [comment]
I'm reading https://www.programmersought.com/article/34764541848/

image


Then I  want to add 
image

Is it necessary?
@AnirudhDagar 

My test here: https://github.com/StevenJokes/d2l-en-read/blob/master/chapter-generative-adversarial-networks/PR1308%20copy%202.ipynb

**Comment by mli** [comment]
Job d2l-en/PR-1358/10 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1358/

**Comment by mli** [comment]
Job d2l-en/PR-1358/16 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1358/

**Comment by StevenJokes** [comment]
@AnirudhDagar 
Please review again. Sorry for too many PR to disturb you.

**Comment by AnirudhDagar** [comment]
Thanks, @StevenJokes I'm merging this now and will make a few changes to fix a couple of issues.

**Review by AnirudhDagar — APPROVED** [review]

**Comment by AnirudhDagar** [comment]
Please see #1400 for the fixes!

**Pull request merged**